baseapp/baseapp_test.go

```diff
@@ -1821,17 +1822,35 @@ func TestListSnapshots(t *testing.T) {
 	require.NoError(t, err)
 	defer teardown()
 
+	expected := abci.ResponseListSnapshots{Snapshots: []*abci.Snapshot{
+		{Height: 4, Format: 1, Chunks: 2},
+		{Height: 2, Format: 1, Chunks: 1},
+	}}
+
 	resp := app.ListSnapshots(abci.RequestListSnapshots{})
-	for _, s := range resp.Snapshots {
+	queryResponse := app.Query(abci.RequestQuery{
+		Path: "/app/snapshots",
+	})
+
+	queryListSnapshotsResp := abci.ResponseListSnapshots{}
+	err = json.Unmarshal(queryResponse.Value, &queryListSnapshotsResp)
+	require.NoError(t, err)
+
+	for i, s := range resp.Snapshots {
+		querySnapshot := queryListSnapshotsResp.Snapshots[i]
+
 		assert.NotEmpty(t, s.Hash)
 		assert.NotEmpty(t, s.Metadata)
 		s.Hash = nil
 		s.Metadata = nil
+
+		assert.NotEmpty(t, querySnapshot.Hash)
+		assert.NotEmpty(t, querySnapshot.Metadata)
+		querySnapshot.Hash = nil
+		querySnapshot.Metadata = nil
```

Comment:
```suggestion
		assert.Equal(t, *s, querySnapshot)
		assert.NotEmpty(t, s.Hash)
		assert.NotEmpty(t, s.Metadata)
		s.Hash = nil
		s.Metadata = nil
```

Reply:
check that query snapshot is the same as other 'actual' snapshot. Then mutate s for the expected check to work